test/controllers/transactions_controller_test.rb

```diff
@@ -138,4 +142,40 @@ class TransactionsControllerTest < ActionDispatch::IntegrationTest
     assert_redirected_to transactions_url
     assert_enqueued_with(job: AccountSyncJob)
   end
+
+  test "can destroy many transactions at once" do
+    delete_count = 10
+    assert_difference("Transaction.count", -delete_count) do
+      post bulk_delete_transactions_url, params: { bulk_delete: { transaction_ids: @recent_transactions.first(delete_count).pluck(:id) } }
+    end
+
+    assert_redirected_to transactions_url
+    assert_equal "10 transactions deleted", flash[:notice]
+  end
+
+  test "can update many transactions at once" do
+    transactions = @user.family.transactions.ordered.limit(20)
+
+    transactions.each do |transaction|
+      transaction.update! excluded: false, currency: "USD", category_id: Transaction::Category.first.id
+    end
+
+    post bulk_update_transactions_url, params: {
+      bulk_update: {
+        transaction_ids: transactions.map(&:id),
+        excluded: true,
+        currency: "CAD",
+        category_id: Transaction::Category.second.id
+      }
+    }
+
+    assert_redirected_to transactions_url
+    assert_equal "#{transactions.count} transactions updated", flash[:notice]
+
+    transactions.reload.each do |transaction|
+      assert transaction.excluded
+      assert_equal "CAD", transaction.currency
+      assert_equal Transaction::Category.second, transaction.category
+    end
+  end
```

Comment:
The next PR will implement these bulk updates.  Leaving this here as a placeholder.